### PR TITLE
Mr/user stats

### PR DESCRIPTION
**Pull request by mronauli**
## Description
add functionality for user to see its stats... currently the count of how many posts they've made

## Notes
testing for edge case that a user's posts from other mods can also be accounted for added to card on asana

## RSpec Results
```
Finished in 0.60429 seconds (files took 3.47 seconds to load)
24 examples, 0 failures

Coverage report generated for RSpec to /Users/mariaronauli/turing/3module/projects/ghost-diary-gateway/coverage. 168 / 168 LOC (100.0%) covered.
```

### CODE REVIEW

File: spec/features/users/user_can_see_its_stats_page_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      group_1 = Group.create(name: 'Mod 1')
+      user = User.create(name: 'user', uid: '1111', group: group_1)
+      day_1 = Day.create(week: 1, day_of_week: 1, group: group_1)
+      day_2 = Day.create(week: 2, day_of_week: 5, group: group_1)
+      day_3 = Day.create(week: 4, day_of_week: 3, group: group_1)
+      day_4 = Day.create(week: 6, day_of_week: 4, group: group_1)
+      post_1 = Post.create(body: "Wow, first day of the mod 1... syllabus day... but still so lost @.@", tone: "confuse", day: day_1, user: user)
+      post_2 = Post.create(body: "Just took my first in class challenge. I'm NEVER going to get any better at this!!", tone: "angry", day: day_2, user: user)
+      post_3 = Post.create(body: "Learning that I gotta utilize my resources and getting better doesn't come out of thin air. Make sure to use paired.tech!!", tone: "optimistic", day: day_3, user: user)
+      post_4 = Post.create(body: "Passed Mod 1!! Time to get schwifty with the Turing crew", tone: "excited", day: day_4, user: user)
```
Comment (d-atkins) [inline spec/features/users/user_can_see_its_stats_page_spec.rb:15]: I don't think any of these need to be saved as variables

File: spec/features/users/user_can_see_its_stats_page_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
+      visit "/profile/stats"
+
+      expect(page).to have_content("Total posts:")
```
Comment (d-atkins) [inline spec/features/users/user_can_see_its_stats_page_spec.rb:20]:
```suggestion
      expect(page).to have_content("Total posts: 4")
```

File: spec/features/users/user_can_see_its_stats_page_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+require "rails_helper"
```
Comment (d-atkins) [inline spec/features/users/user_can_see_its_stats_page_spec.rb:1]: I would change the path/filename of this test to the following:
`spec/features/users/profile/stats/index_spec.rb`

File: spec/features/users/user_can_see_its_stats_page_spec.rb (hunk 2 above)
Comment (d-atkins) [reply spec/features/users/user_can_see_its_stats_page_spec.rb]: by the way, be careful with the commit suggestion button. if you use it, make sure your local and remote branches are the same, and then pull down the updated remote after committing via github

Comment (mronauli) [reply spec/features/users/user_can_see_its_stats_page_spec.rb]: duly noted!! will watch out for that

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -13,4 +13,20 @@
     it { should have_many(:posts) }
     it { should belong_to(:group) }
   end
+
+  describe 'post_count' do
```
Comment (d-atkins) [inline spec/models/user_spec.rb:17]:
```suggestion
  describe 'methods' do
```

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -13,4 +13,20 @@
...
+    it "can count the amount of posts a user has made" do
```
Comment (d-atkins) [inline spec/models/user_spec.rb:18]:
```suggestion
    it 'post_count' do
```

**Pull request merged**